settings.gradle

```diff
@@ -6,6 +6,7 @@ include 'junit-jupiter-engine'
 include 'junit-jupiter-migrationsupport'
 include 'junit-jupiter-params'
 include 'junit-platform-commons'
+if (JavaVersion.current().isJava9Compatible()) include 'junit-platform-commons-java-9'
```

Comment:
Shall we make this mandatory right away?

Reply:
And ditch JDK 8?

Reply:
Well, we'll have to build on JDK 9 once we merge this PR to build correct artifacts, don't we?

Reply:
True.

The major source and target compatibility stays on `1.8`. Only selected subprojects need `1.9`. Mh, should work.

Reply:
Jenkins CI builds need to be suspended until they support JDK 9.

Reply:
Travis CI build matrix can be reduced to JDK 9 only.

Reply:
`junit5-samples` should stay as they are, meaning most of the sample run on JDK 8 and only some on JDK 9.

Reply:
I just opened a support request with CloudBees to get JDK 9 support. Let's hope they respond soon.

Reply:
Nit: current IDE support for JDK-9 is still moving. E.g. IDEA sometimes fails to "Refresh all Gradle projects" with a "MODULE" not supported error. Eclipse might have similar ... challenges.

Reply:
See https://stackoverflow.com/questions/45899027/errorbug-exception-in-phase-class-generation-in-source-unit-buildscript

Reply:
When we do switch to JDK-9 as our build platform, we should also provide _real_ `module-info.class` as `--release 9` assets. At least for `org.junit.platform.commons` to export only those packages we want to used by external code.

Reply:
Opened https://github.com/junit-team/junit5/issues/1091 to address the _real_ Module Descriptor.

Reply:
When #1097 is merged, I'll rebase this PR and remove the `if (JavaVersion.current().isJava9Compatible())` condition.